### PR TITLE
Remove the deprecated `agg_and_log_metrics`

### PR DESCRIPTION
**Pull request by krishnakalyan3**
## What does this PR do?
Removes deprecated API

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -43,10 +43,6 @@ class LightningLoggerBase(logger.Logger):
         .. deprecated:: v1.6
             The parameters `agg_key_funcs` and `agg_default_func` are deprecated
             in v1.6 and will be removed in v1.8.
```
Comment (otaj) [inline src/pytorch_lightning/loggers/base.py:45]: The whole `args` section could be removed

File: src/pytorch_lightning/loggers/logger.py [hunk 2]
```diff
@@ -71,10 +71,6 @@ class Logger(ABC):
         .. deprecated:: v1.6
             The parameters `agg_key_funcs` and `agg_default_func` are deprecated
             in v1.6 and will be removed in v1.8.
```
Comment (otaj) [inline src/pytorch_lightning/loggers/logger.py:73]: The whole `args` section could be removed

**Review by carmocca — APPROVED** [review]
I finished up the removal of the aggregation logic

**Review by otaj — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by krishnakalyan3** [comment]
Thank you @carmocca

**Pull request merged**